File: CS/CS/Controllers/HomeController.cs
using System;
using System.Web.Mvc;
using CS.Models;

namespace CS.Controllers {
    public class HomeController : Controller {
        public ActionResult Index() {
            if(Session["TypedListModel"] == null)
                Session["TypedListModel"] = InMemoryModel.GetTypedListModel();
            
            return View(Session["TypedListModel"]);
        }

        public ActionResult TypedListDataBindingPartial() {
            return PartialView(Session["TypedListModel"]);
        }
    }
}

File: VB/CS/Controllers/HomeController.vb
Imports System.Web.Mvc
Imports CS.Models

Namespace CS.Controllers

    Public Class HomeController
        Inherits Controller

        Public Function Index() As ActionResult
            If Session("TypedListModel") Is Nothing Then Session("TypedListModel") = InMemoryModel.GetTypedListModel()
            Return View(Session("TypedListModel"))
        End Function

        Public Function TypedListDataBindingPartial() As ActionResult
            Return PartialView(Session("TypedListModel"))
        End Function
    End Class
End Namespace
